block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/SubscriberServicePlugin.java

```diff
@@ -0,0 +1,255 @@
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import com.swirlds.metrics.api.LongGauge;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.lang.System.Logger;
+import java.lang.System.Logger.Level;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentSkipListMap;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorCompletionService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.atomic.AtomicLong;
+import org.hiero.block.api.SubscribeStreamRequest;
+import org.hiero.block.api.SubscribeStreamResponse;
+import org.hiero.block.internal.SubscribeStreamResponseUnparsed;
+import org.hiero.block.node.spi.BlockNodeContext;
+import org.hiero.block.node.spi.BlockNodePlugin;
+import org.hiero.block.node.spi.ServiceBuilder;
+
+/** Provides implementation for the health endpoints of the server. */
+public class SubscriberServicePlugin implements BlockNodePlugin, ServiceInterface {
+    /** The logger for this class. */
+    private final Logger LOGGER = System.getLogger(getClass().getName());
+    /** The block node context */
+    private BlockNodeContext context;
+
+    private SubscriberConfig pluginConfiguration;
+    /** Metric for the number of subscribers receiving block items. */
+    private final List<SubscribeBlockStreamHandler> clientHandlers = new LinkedList<>();
+
+    private SubscribeBlockStreamHandler clientHandler;
+
+    /*==================== BlockNodePlugin Methods ====================*/
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void init(BlockNodeContext context, ServiceBuilder serviceBuilder) {
+        this.context = context;
+        pluginConfiguration = context.configuration().getConfigData(SubscriberConfig.class);
+        // register us as a service
+        serviceBuilder.registerGrpcService(this);
+    }
+
+    /*==================== ServiceInterface Methods ====================*/
+
+    /**
+     * BlockStreamSubscriberService types define the gRPC methods available on the BlockStreamSubscriberService.
+     */
+    enum SubscriberServiceMethod implements Method {
+        /**
+         * The subscribeBlockStream method represents the server-streaming gRPC method
+         * consumers should use to subscribe to the BlockStream from the Block Node.
+         */
+        subscribeBlockStream
+    }
+
+    @Override
+    public void start() {
+        // Create the client handler and wait for it to start and reach a ready state.
+        clientHandler = new SubscribeBlockStreamHandler(context, this);
+    }
+
+    @Override
+    public void stop() {
+        clientHandler.stop();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @NonNull
+    public String serviceName() {
+        return "BlockStreamService";
```

Comment:
```suggestion
        String[] parts = fullName().split("\\.");
        return parts[parts.length - 1];
```

Otherwise it fails with 👇 , when you try to connect consumer

```
2025-04-25 16:12:39.973+0300 SEVERE  [org.hiero.block.simulator.grpc.impl.ConsumerStreamObserver onError] Error io.grpc.StatusRuntimeException: UNKNOWN: HTTP status code 200
invalid content-type: null
trailers: Metadata(:status=200,grpc-status=5) with status Status{code=UNKNOWN, description=HTTP status code 200
invalid content-type: null
trailers: Metadata(:status=200,grpc-status=5), cause=null}.
io.grpc.StatusRuntimeException: UNKNOWN: HTTP status code 200
invalid content-type: null
trailers: Metadata(:status=200,grpc-status=5)
```
Change fixes this, it was discovered in this [PR](https://github.com/hiero-ledger/hiero-block-node/pull/891)